File: app/src/main/java/com/corphish/nightlight/receivers/BootCompleteReceiver.java
package com.corphish.nightlight.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import com.corphish.nightlight.helpers.BootUtils;
import com.corphish.nightlight.services.BootCompleteJobService;

/**
 * Created by Avinaba on 10/5/2017.
 * Broadcast listener for boot completion
 */

public class BootCompleteReceiver extends BroadcastReceiver {
    private static final String BOOT_COMPLETE_ANDROID_STRING = "android.intent.action.BOOT_COMPLETED";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (BOOT_COMPLETE_ANDROID_STRING.equals(intent.getAction())) return;

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) BootCompleteJobService.schedule(context);
        else BootUtils.applyOnBoot(context);
    }
}

File: app/src/main/java/com/corphish/nightlight/receivers/BootCompleteReceiver.kt
package com.corphish.nightlight.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import com.corphish.nightlight.data.Constants

import com.corphish.nightlight.helpers.BootUtils
import com.corphish.nightlight.helpers.PreferenceHelper
import com.corphish.nightlight.services.BootCompleteJobService

/**
 * Created by Avinaba on 10/5/2017.
 * Broadcast listener for boot completion
 */

class BootCompleteReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED != intent.action) return
        if (!PreferenceHelper.getBoolean(context, Constants.PREF_SET_ON_BOOT, Constants.DEFAULT_SET_ON_BOOT)) {
            PreferenceHelper.putBoolean(context, Constants.PREF_FORCE_SWITCH, false)

            return
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
            BootCompleteJobService.schedule(context)
        else
            BootUtils.applyOnBoot(context)
    }
}
